examples/mcp-client/src/server.ts

```diff
@@ -58,12 +56,22 @@ export class MyAgent extends Agent<Env, State> {
 
   async addMcpServer(url: string): Promise<string> {
     console.log(`Registering server: ${url}`);
-    const { id, authUrl } = await this.mcp.connect(url);
+    const authProvider = new DurableObjectOAuthClientProvider(
+      this.ctx.storage,
+      this.name,
+      `${this.env.HOST}/agents/my-agent/${this.name}/callback`
+    );
+    const { id, authUrl } = await this.mcp.connect(url, {
+      transport: { authProvider },
```

Comment:
should we pass this here, or in the constructor? what benefit to either approach?

Reply:
Each transport is intended to be fully owned by the client and not shared. Because of that, I designed the auth provider to be owned by an individual client, so it ends up being pass in during the `connect`. Technically, there's no reason you couldn't share auth providers between client connections, but I think this way maps a bit better to how the transport is set up.